src/main/java/uk/gov/pay/api/model/links/PaymentWithAllLinks.java

```diff
@@ -100,4 +101,201 @@ public static PaymentWithAllLinks getPaymentWithLinks(
         
         return PaymentWithAllLinks.valueOf(paymentConnector, selfLink, paymentEventsUri, paymentCancelUri, paymentRefundsUri, paymentsCaptureUri, paymentAuthorisationUri);
     }
+    
+    public static class PaymentWithAllLinksBuilder {
+        private String chargeId;
+        private long amount;
+        private PaymentState state;
+        private String returnUrl;
+        private String description;
+        private String reference;
+        private String email;
+        private String paymentProvider;
+        private String createdDate;
+        private SupportedLanguage language;
+        private boolean delayedCapture;
+        private boolean moto;
+        private RefundSummary refundSummary;
+        private PaymentSettlementSummary settlementSummary;
+        private CardDetails cardDetails;
+        private List<PaymentConnectorResponseLink> paymentConnectorResponseLinks;
+        private URI selfLink;
+        private URI paymentEventsUri;
+        private URI paymentCancelUri;
+        private URI paymentRefundsUri;
+        private URI paymentCaptureUri;
+        private URI paymentAuthorisationUri;
+        private Long corporateCardSurcharge;
+        private Long totalAmount;
+        private String providerId;
+        private ExternalMetadata metadata;
+        private Long fee;
+        private Long netAmount;
+        private AuthorisationSummary authorisationSummary;
+        private String agreementId;
+        private AuthorisationMode authorisationMode;
+
+        public PaymentWithAllLinksBuilder setChargeId(String chargeId) {
```

Comment:
the builder pattern usually has method prefix `with...` in our repos

Reply:
ah...does it really matter? This was auto-generated by Intellij...

Reply:
Usually it lets you select which one to use doesn't it? or maybe I have a specific plugin installed to generate builders

Consistency is nice just because if you expect one thing because its like that everywhere else but its different, it's just a slight annoyance in your day.

Reply:
Ah I have this plugin installed https://plugins.jetbrains.com/plugin/6585-builder-generator